### ISSUE TITLE
Wrap each response with the Response object

### ISSUE THREAD

**Issue by veyndan** [Meisolsson/GitHubSdk#8]
By wrapping each response with a `Response` object, an exception won't be thrown if there is a network error, but will be passed to `onSuccess` and we can handle whether it was a successful response or not. An HTTP exception (I believe) doesn't warrant a call to `onError` as it is an exception that can be expected (this is talked about in the first link if you rewind back a minute from where a timestamped it).

Dan Lew talks about it [here](https://youtu.be/QdmkXL7XikQ?t=35m40s) and [here](http://disq.us/p/tt2ahe) (in the last paragraph).

**Comment by veyndan** [comment]
Any thoughts on this?

**Comment by Meisolsson** [comment]
I think the idea is good, but there's one thing i dislike. Since a lot of calls use Page<T> they will look bad being wrapped in `Response` (example `Response<Page<NotificationThread>>`). 

Solving that is actually quite easy, since Response gives us the headers we can just give the users a static method somewhere to fetch a page object (or just the page value they need) from the Response.headers().

So basically if you do want to do this, don't use `Page` as a response and remove the page values from the `SearchPage`.

**Comment by veyndan** [comment]
It isn't often that we'd see `Response<Page<NotificationThread>>` as lambdas are usually used so the object type wouldn't be visible.

I'm not totally sure what you mean by your solution, do you mind clarifying it perhaps with some pseudocode? I don't understand how you'd retrieve the `Page` object from `Response.headers()` nor how the user API would look like.

**Comment by Meisolsson** [comment]
Actually forget what I said, we would still need to replace `Page` with `List`. So just wrap all of the responses with `Response`
